src/envoy/mixer/stats.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/mixer/stats.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Http {
+namespace Mixer {
+namespace {
+
+const int kStatsUpdateIntervalInMs = 10000;
+
+}  // namespace
+
+MixerStatsObject::MixerStatsObject(Event::Dispatcher& dispatcher,
+                                   const std::string& name, Stats::Scope& scope)
+    : stats_{ALL_MIXER_FILTER_STATS(POOL_COUNTER_PREFIX(scope, name))},
+      dispatcher_(dispatcher) {}
+
+::istio::mixer_client::Statistics* MixerStatsObject::mutate_old_stats() {
+  return &old_stats_;
+}
+
+void MixerStatsObject::InitGetStatisticsFunc(GetStatsFunc get_stats) {
+  if (get_stats) {
+    get_stats_func_ = get_stats;
+  } else {
+    auto& logger = Logger::Registry::getLog(Logger::Id::config);
+    ENVOY_LOG_TO_LOGGER(logger, error, "get_stats is empty");
+  }
+}
+
+void MixerStatsObject::GetStatistics(::istio::mixer_client::Statistics* stats) {
+  if (get_stats_func_) {
+    get_stats_func_(stats);
+  } else {
+    auto& logger = Logger::Registry::getLog(Logger::Id::config);
```

Comment:
at constructor, if get_stats_func_ is null, don't create timer.  not need for this check